src/main/java/prefs/UserConfig.java

```diff
@@ -0,0 +1,7 @@
+package prefs;
+
+/**
+ * Abstraction for user-defined settings.
+ */
+public class UserConfig {
+}
```

Comment:
The preferences class needs to be adapted to take in the user config as well. And after this PR, all other PRs can use this class to store their user-specific info, instead of each creating their own userconfig class and encountering difficult merge issues